external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpoutConfig.java

```diff
@@ -116,217 +141,57 @@
-            this.valueDes = valueDes;
-            this.valueDesClazz = valueDesClazz;
-        }
-
-        /**
-         * Specifying this key deserializer overrides the property key.deserializer. If you have
-         * set a custom RecordTranslator before calling this it may result in class cast
-         * exceptions at runtime.
-         */
-        public <NewKeyT> Builder<NewKeyT,V> setKey(SerializableDeserializer<NewKeyT> keyDeserializer) {
-            return new Builder<>(this, keyDeserializer, null, valueDes, valueDesClazz);
-        }
-        
-        /**
-         * Specify a class that can be instantiated to create a key.deserializer
-         * This is the same as setting key.deserializer, but overrides it. If you have
-         * set a custom RecordTranslator before calling this it may result in class cast
-         * exceptions at runtime.
-         */
-        public <NewKeyT> Builder<NewKeyT, V> setKey(Class<? extends Deserializer<NewKeyT>> clazz) {
-            return new Builder<>(this, null, clazz, valueDes, valueDesClazz);
-        }
-
-        /**
-         * Specifying this value deserializer overrides the property value.deserializer.  If you have
-         * set a custom RecordTranslator before calling this it may result in class cast
-         * exceptions at runtime.
-         */
-        public <NewValueT> Builder<K,NewValueT> setValue(SerializableDeserializer<NewValueT> valueDeserializer) {
-            return new Builder<>(this, keyDes, keyDesClazz, valueDeserializer, null);
+            this.translator = new DefaultRecordTranslator<>();
         }
         
         /**
-         * Specify a class that can be instantiated to create a value.deserializer
-         * This is the same as setting value.deserializer, but overrides it.  If you have
-         * set a custom RecordTranslator before calling this it may result in class cast
-         * exceptions at runtime.
-         */
-        public <NewValueT> Builder<K,NewValueT> setValue(Class<? extends Deserializer<NewValueT>> clazz) {
-            return new Builder<>(this, keyDes, keyDesClazz, null, clazz);
-        }
-        
-        /**
-         * Set a Kafka property config.
+         * Set a Kafka consumer property config. 
          */
         public Builder<K,V> setProp(String key, Object value) {
             kafkaProps.put(key, value);
             return this;
         }
         
         /**
-         * Set multiple Kafka property configs.
+         * Set multiple Kafka consumer property configs.
          */
         public Builder<K,V> setProp(Map<String, Object> props) {
             kafkaProps.putAll(props);
             return this;
         }
         
         /**
-         * Set multiple Kafka property configs.
+         * Set multiple Kafka consumer property configs.
          */
         public Builder<K,V> setProp(Properties props) {
             for (String name: props.stringPropertyNames()) {
```

Comment:
This code has one bug as it only picks properties for which both key and value are both strings. It should also pick key/value pairs that are String/Object. I suggest that we refactor this code to be as such:

```java
    public Builder<K,V> setProp(Properties props) {
            props.forEach((key, value) -> {
                if (key instanceof String) {
                    kafkaProps.put((String) key, value);
                }
            });
            return this;
        }
```